odoo/orm/fields_relational.py

```diff
@@ -746,12 +758,22 @@ def update_db(self, model, columns):
                     comodel=self.comodel_name
                 ))
 
-    def get_domain_list(self, records):
-        comodel = records.env.registry[self.comodel_name]
-        inverse_field = comodel._fields[self.inverse_name]
-        domain = super(One2many, self).get_domain_list(records)
-        if inverse_field.type == 'many2one_reference':
-            domain = domain + [(inverse_field.model_field, '=', records._name)]
+    def get_comodel_domain(self, model) -> Domain:
+        domain = super().get_comodel_domain(model)
+        if self.comodel_name and self.inverse_name:
+            comodel = model.env.registry[self.comodel_name]
+            inverse_field = comodel._fields[self.inverse_name]
+            if inverse_field.type == 'many2one_reference':
+                domain &= Domain(inverse_field.model_field, '=', model._name)
```

Comment:
Note for another PR: I see a lot of One2many contains `lambda self: [('model', '=', self._name), ...` where the Many2oneReference. We should make a small PR to remove it to avoid useless computation.

Reply:
Ok, I think we could event redefine a new field type for references.